### ISSUE TITLE
Zero value in the middle of padded input of Embedding layer

### ISSUE THREAD

**Issue by pntuananh** [keras-team/keras#8150]
Can the zero values be in the middle of the input of Embedding layer using padding mask? For example input = [1,2,0.3,4,0,0...]. For some reasons, I do not want to right pad or left pad my input. I want the zero values to be in the middle of the input. Is it possible for Embedding layer of Keras?

**Comment by tRosenflanz** [comment]
There is no direct way to implement this. Given a list [1,2,3,4] it is impossible to infer where you want 0s to be inserted specifically. You are better off writing your own numpy based function to do that

**Comment by pntuananh** [comment]
@tRosenflanz : Perhaps I did not state my question clearly. For Embedding layer using padding, we have to give it the vector input (containing IDs) with zeros either at the end (right-padding) or at the beginning (left-padding). However, in my model, I want zeros can also be in the middle of the input vector, for example, input vector = [3,2,0,4,0,2,...]. I'm not sure if it is possible in Keras. I tried to search but could not find any answer.
